src/FAudioFX.c

```diff
@@ -137,30 +138,267 @@ static FAPORegistrationProperties ReverbProperties =
+		FAPOParametersBase_BeginProcess(&fapo->base);*/
+	FAudioFXReverbTestParameters *params = (FAudioFXReverbTestParameters*)
 		FAPOParametersBase_BeginProcess(&fapo->base);
 
-	/* TODO */
-	(void) params;
+	const float *input_buffer = (const float *)pInputProcessParameters->pBuffer;
+	float *output_buffer = (float *) pOutputProcessParameters->pBuffer;
+
+	/* update parameters (FIXME: only if necessary?) */
+	RoomReverb *reverb = &fapo->network;
+	FAudioFXFilterDelay_Change(reverb->early_delay, (float) params->ReflectionsDelay, 0.0f);
+
+
+	float early_diffusion = 0.6f - ((params->EarlyDiffusion / 15.0f) * 0.2f);
+	FAudioFXFilterAllPass_Change(reverb->apf_in[0], params->InDiffusionLength1, early_diffusion);
+	FAudioFXFilterAllPass_Change(reverb->apf_in[1], params->InDiffusionLength2, -early_diffusion);
+
+	uint32_t delay = (params->ReverbDelay > params->ReflectionsDelay) ? params->ReverbDelay - params->ReflectionsDelay : params->ReverbDelay;
+	FAudioFXFilterDelay_Change(reverb->reverb_delay, (float) delay, 0.0f);
+
+	float comb_delays[] = { 25.31f, 26.94f, 28.96f, 30.75f, 32.24f, 33.8f, 35.31f, 36.67f };
+	for (int32_t comb = 0; comb < 8; ++comb)
+	{
+		FAudioFXFilterCombShelving_Change(reverb->lpf_comb[comb], comb_delays[comb], params->DecayTime * 1000.0f * 0.2f);
+
+		FAudioFXFilterBiQuad_Change(
+			&reverb->lpf_comb[comb]->low_shelving,
+			50.0f + params->LowEQCutoff * 50.0f,
+			0.0f,
+			params->LowEQGain - 8.0f);
+		FAudioFXFilterBiQuad_Change(
+			&reverb->lpf_comb[comb]->high_shelving,
+			1000 + params->HighEQCutoff * 500.0f,
+			0.0f,
+			params->HighEQGain - 8.0f);
+	}
+	
+	reverb->early_gain = FAudioFX_INTERNAL_DbGainToFactor(params->ReflectionsGain);
+	reverb->reverb_gain = FAudioFX_INTERNAL_DbGainToFactor(params->ReverbGain);
+
+	reverb->room_gain = FAudioFX_INTERNAL_DbGainToFactor(params->RoomFilterMain);
+	FAudioFXFilterBiQuad_Change(fapo->network.room_high_shelf, params->RoomFilterFreq, 0.0f, params->RoomFilterMain + params->RoomFilterHF);
+
+	float late_diffusion = 0.6f - ((params->LateDiffusion / 15.0f) * 0.2f);
+	float allpass_delays[] = { 5.10f, 12.61f, 10.0f, 7.73f };
+
+	for (int32_t ap = 0; ap < 4; ++ap)
+	{
+		FAudioFXFilterAllPass_Change(reverb->apf_out[ap], allpass_delays[ap], late_diffusion);
+	}
+
+	reverb->wet_ratio = params->WetDryMix / 100.0f;
+	reverb->dry_ratio = 1.0f - reverb->wet_ratio;
+
+	/* run the reverberation effect */
+	for (size_t idx = 0; idx < pInputProcessParameters->ValidFrameCount; ++idx)
+	{
+		float in = input_buffer[idx];
+
+		/* pre delay */
+		float delay_in = FAudioFXFilterDelay_Process(reverb->early_delay, in);
+
+		/* early reflections */
+		float early = FAudioFXFilterAllPass_Process(reverb->apf_in[0], delay_in);
+		early = FAudioFXFilterAllPass_Process(reverb->apf_in[0], early);
+
+		early = early * reverb->early_gain;
+
+		/* reverberation */
+		float revdelay = FAudioFXFilterDelay_Process(reverb->reverb_delay, early);
+		float comb_out = 0.0f;
+		float gain = 0.125f;
```

Comment:
I notice that if I set this to 1.0f it sounds pretty good with our test case, but I don't know if there's something important to this value...